src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +91,211 @@ def _get_symbol_table(node):
+            "method _check_args 'dim' argument should be at "
+            "most the number of dimensions of the array '{0}' but found "
+            "'{1}'.".format(len(array.children), dim))
+
+    # The first child of the array (index 0) relates to the first
+    # dimension (dim 1), so we need to reduce dim by 1.
+    if not isinstance(array.children[dim-1], Range):
+        raise TypeError(
+            "method _check_args 'array' argument index '{0}' "
+            "should be a Range type but found '{1}'."
+            "".format(dim-1, type(array.children[dim-1]).__name__))
+
+
+def _is_bound_full_extent(array, dim, operator):
+    '''A Fortran array section with a missing lower bound implies the
+    access starts at the first element and a missing upper bound
+    implies the access ends at the last element e.g. a(:,:)
+    accesses all elements of array a and is equivalent to
+    a(lbound(a,1):ubound(a,1),lbound(a,2):ubound(a,2)). The PSyIR
+    does not support the shorthand notation, therefore the lbound
+    and ubound operators are used in the PSyIR.
+
+    This utility function checks that shorthand lower or upper
+    bound Fortran code is captured as longhand lbound and/or
+    ubound functions as expected in the PSyIR.
+
+    The supplied "array" argument is assumed to be an Array node and
+    the contents of the specified dimension "dim" is assumed to be a
+    Range node.
+
+    This routine is only in fparser2.py until #717 is complete as it
+    is used to check that array syntax in a where statement is for the
+    full extent of the dimension. Once #717 is complete this routine
+    can be moved into fparser2_test.py as it is used there in a
+    different context.
+
+    :param array: the node to check.
+    :type array: :py:class:`pysclone.psyir.node.array`
+    :param int dim: the dimension index to use.
+    :param operator: the operator to check.
+    :type operator: \
+        :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.LBOUND` \
+        or :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.UBOUND`
+
+    :raises TypeError: if the supplied arguments are of the wrong type.
+    :raises NotImplementedError: if the supplied array does not have \
+        the expected properties.
+
+    '''
+    _check_args(array, dim)
+
+    if operator == BinaryOperation.Operator.LBOUND:
+        index = 0
+    elif operator == BinaryOperation.Operator.UBOUND:
+        index = 1
+    else:
+        raise TypeError(
+            "'operator' argument  expected to be LBOUND or UBOUND but "
+            "found '{0}'.".format(type(operator).__name__))
+
+    # The first child of the array (index 0) relates to the first
+    # dimension (dim 1), so we need to reduce dim by 1.
+    bound = array.children[dim-1].children[index]
+
+    if not isinstance(bound, BinaryOperation):
+        return False
+
+    reference = bound.children[0]
+    literal = bound.children[1]
+
+    if (bound.operator == operator
```

Comment:
pylint complains that there are too many booleans in statement. I think it looks OK though so feel free to temporarily disable this warning.

Reply:
Done.

Reply:
In fact you could change this to just:

    return (bound.operator == operator and isinstance(reference, ......)

and remove the `return False`